app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/api/components/ApiKeyButton.tsx

```diff
@@ -25,43 +21,23 @@ export const ApiKeyButton = ({
   const { id } = useParams();
 
   const { Event } = useCustomEvent();
-  const [pendingKey, setPendingKey] = useState<ApiKeyType | null>(null);
-  const [generatingKey, setGeneratingKey] = useState(false);
-  const [deletingKey, setDeletingKey] = useState(false);
-
-  if (Array.isArray(id)) return null;
 
   const openDialog = () => {
-    Event.fire(EventKeys.openApiKeyDialog, {
-      download: () => {
-        downloadKey(JSON.stringify(pendingKey), id);
-        setPendingKey(null);
-      },
-      cancel: async () => {
-        setPendingKey(null);
-        setDeletingKey(true);
-        // Note this will revalidate the page
-        // and remove the key from the UI
-        await deleteServiceAccountKey(id);
-        setDeletingKey(false);
-      },
-    });
+    Event.fire(EventKeys.openApiKeyDialog, { id });
   };
 
+  if (Array.isArray(id)) return null;
+
   return (
     <div className="mb-4">
-      {showDelete && keyId && !deletingKey && !pendingKey ? (
+      {showDelete && keyId ? (
         <DeleteKeyButton id={id} keyId={keyId} />
       ) : (
         <GenerateApiKeyButton
-          loading={generatingKey}
+          loading={false}
           theme="primary"
-          disabled={Boolean(keyId) || pendingKey !== null}
+          disabled={Boolean(keyId)}
           onClick={async () => {
```

Comment:
I am guessing we probably don't need the `async` here anymore?

```suggestion
          onClick={() => {
```